proposals/NNNN-task-executor-preference.md

```diff
@@ -0,0 +1,664 @@
+
+## Proposed solution
+
+We propose to introduce an additional layer of control over where a task can be executed, and have this executor setting be “sticky” to the task.
+
+**Currently** the decision where an async function or closure is going to execute is binary:
+
+```
+// `func` execution semantics before this proposal
+
+[ func / closure ] - /* where should it execute? */
+                                 |
+                           +--------------+          +==========================+
+                   +- no - | is isolated? | - yes -> | default (actor) executor |
+                   |       +--------------+          +==========================+
+                   |
+                   |                                 +==========================+
+                   +-------------------------------> | on global conc. executor |
+                                                     +==========================+
+```
+
+This proposal introduces a way to control hopping off to the global concurrent pool for `nonisolated` functions and closures. This is expressed as **task executor preference** and is sticky to the task and entire *structured task hierarchy* created from a task with a specified preference. This changes the current decision diagram to the following:
+
+```
+// `func` execution semantics with this proposal
+
+[ func / closure ] - /* where should it execute? */
+                               |
+                     +--------------+          +===========================+
+           +--- no - | is isolated? | - yes -> | actor has unownedExecutor |
+           |         +--------------+          +===========================+
+           |                                       |                |      
+           |                                      yes               no
+           |                                       |                |
+           |                                       v                v
+           |                  +=======================+    /* task executor preference? */
+           |                  | on specified executor |        |                   |
+           |                  +=======================+       yes                  no
+           |                                                   |                   |
+           |                                                   |                   v
+           |                                                   |    +==========================+
+           |                                                   |    | default (actor) executor |
+           |                                                   v    +==========================+
+           v                                   +==============================+
+/* task executor preference? */ ---- yes ----> | on Task's preferred executor |
+           |                                   +==============================+
+          yes
```

Comment:
I'm pretty sure you want to use `no` in this case:
```suggestion
           no
```

Reply:
nice catch, thanks